app/navigation/mainStackNavigator.tsx

```diff
@@ -0,0 +1,107 @@
+import {createNativeStackNavigator} from '@react-navigation/native-stack';
+import {createDrawerNavigator} from '@react-navigation/drawer';
+import {CartIcon} from '../components/cartIcon';
+import {Text, View} from 'react-native';
+import {styles} from '../screens/mainPage/styles/mainStyles';
+import Icon from 'react-native-vector-icons/FontAwesome';
+import MainScreen from '../screens/mainPage/MainScreen';
+import {TestScreen} from '../screens/testScreen';
+import ProductDetailsScreen from '../screens/productDetails/ProductDetailsScreen';
+import {productDetailsStyles} from '../screens/productDetails/styles/productDetailsStyles';
+import AuthorizedCartScreen from '../screens/cart/authorizedCartScreen';
+import {SelectColorModal} from '../modals/selectColorModal';
+import {LoginToContinueModal} from '../modals/loginToContinueModal';
+import {ProductAddedModal} from '../modals/productAddedModal';
+import React from 'react';
+import nonAuthorizedCartScreen from '../screens/cart/nonAuthorizedCartScreen';
+
+const Stack = createNativeStackNavigator();
+const Drawer = createDrawerNavigator();
+
+export function MainStackNavigator({userToken}: {userToken: string | null}) {
+  return (
+    <Stack.Navigator>
+      <Stack.Group>
+        <Stack.Screen
```

Comment:
pls Place drawer navigator at the very top. And then use Stack navigator as one of its screens